src/transport/tcp.rs

```diff
@@ -70,9 +71,16 @@ pub struct TcpHeader {
     ///the URG control bit set.
     pub urgent_pointer: u16,
     ///Buffer containing the options of the header (note that the data_offset defines the actual length). Use the options() method if you want to get a slice that has the actual length of the options.
+    #[cfg_attr(feature = "serde", serde(skip), serde(default = "default_options_buffer"))]
```

Comment:
I think a manual implementation of `Serialize` & `Deserialize` is needed for `TcpHeader` as well. Only the parts of the `options_buffer` that is actually filled should be serialized and/or deserialized. The length of actually filled data in the options_buffer is defined by the `_data_offset` field. The `_data_offset` field itself can be skipped as it is implicitly encoded in the length of the `options_buffer`.

We could also think about instead serializing the decoded options if that is possible. But that is maybe overkill.